service/openstack.py

```diff
@@ -85,13 +88,16 @@ def glance_update_machine_metadata(provider_machine, metadata={}):
             "The method for 'introspecting an image' has changed!"
             " Ask a programmer to fix this!")
     overrides = {
-        "application_version": str(provider_machine.application_version.name),
+        "application_version": str(version.name),
         "application_uuid": base_app.uuid,
         "application_name": _make_safe(base_app.name),
         "application_owner": base_app.created_by.username,
         "application_tags": json.dumps(
             [_make_safe(tag.name) for tag in base_app.tags.all()]),
-        "application_description": _make_safe(base_app.description)}
+        "application_description": _make_safe(base_app.description),
+        "version_name": str(version.name),
```

Comment:
agreed, we should move away from `application_version` (jargon-y) in favor of `version_name`.